packages/module/src/pipelines/components/groups/PipelinesDefaultGroup.tsx

```diff
@@ -0,0 +1,140 @@
+  badge?: string;
+  /** Color to use for the label's badge background */
+  badgeColor?: string;
+  /** Color to use for the label's badge text */
+  badgeTextColor?: string;
+  /** Color to use for the label's badge border */
+  badgeBorderColor?: string;
+  /** Additional classes to use for the label's badge */
+  badgeClassName?: string;
+  /** Location of the badge relative to the label's text, inner or below */
+  badgeLocation?: BadgeLocation;
+  /** Flag if the group is collapsible */
+  collapsible?: boolean;
+  /** Width of the collapsed group */
+  collapsedWidth?: number;
+  /** Height of the collapsed group */
+  collapsedHeight?: number;
+  /** Callback when the group is collapsed */
+  onCollapseChange?: (group: Node, collapsed: boolean) => void;
+  /** Shape of the collapsed group */
+  getCollapsedShape?: (node: Node) => React.FunctionComponent<ShapeProps>;
+  /** Shadow offset for the collapsed group */
+  collapsedShadowOffset?: number;
+  /** Flag if the element selected. Part of WithSelectionProps */
+  selected?: boolean;
+  /** Function to call when the element should become selected (or deselected). Part of WithSelectionProps */
+  onSelect?: OnSelect;
+  /** A ref to add to the node for dragging. Part of WithDragNodeProps */
+  dragNodeRef?: WithDndDragProps['dndDragRef'];
+  /** A ref to add to the node for drag and drop. Part of WithDndDragProps */
+  dndDragRef?: ConnectDragSource;
+  /** A ref to add to the node for dropping. Part of WithDndDropProps */
+  dndDropRef?: ConnectDropTarget;
+  /** Function to call to show a context menu for the node  */
+  onContextMenu?: (e: React.MouseEvent) => void;
+  /** Flag indicating that the context menu for the node is currently open  */
+  contextMenuOpen?: boolean;
+  /** Flag indicating whether to use hull layout or rect layout for expanded groups. Defaults to hull (true) */
+  hulledOutline?: boolean;
+}
+
+type PipelinesDefaultGroupInnerProps = Omit<PipelinesDefaultGroupProps, 'element'> & { element: Node } & WithSelectionProps;
+
+const PipelinesDefaultGroupInner: React.FunctionComponent<PipelinesDefaultGroupInnerProps> = observer(
+  ({ className, element, onCollapseChange, ...rest }) => {
+    const childCount = element.getAllNodeChildren().length;
+    const handleCollapse = (group: Node, collapsed: boolean): void => {
+      if (collapsed && rest.collapsedWidth !== undefined && rest.collapsedHeight !== undefined) {
+        group.setDimensions(new Dimensions(rest.collapsedWidth, rest.collapsedHeight));
+      }
+      group.setCollapsed(collapsed);
+      onCollapseChange && onCollapseChange(group, collapsed);
+    };
+
+    if (element.isCollapsed()) {
+      return (
+        <PipelinesDefaultGroupCollapsed
+          className={className}
+          element={element}
+          onCollapseChange={handleCollapse}
+          badge={`${childCount}`}
+          badgeColor="#f5f5f5"
+          badgeBorderColor="#d2d2d2"
+          badgeTextColor="#000000"
```

Comment:
Use styles here so they can be overridden.

Reply:
Nice catch, thank you @jeff-phillips-18 ! Let me know if these changes are what you had in mind.